client.go

```diff
@@ -3,64 +3,68 @@ package supabase
 import (
 	"errors"
 
+	"github.com/supabase-community/gotrue-go"
+	"github.com/supabase-community/gotrue-go/types"
+	postgrest "github.com/supabase-community/postgrest-go"
 	storage_go "github.com/supabase-community/storage-go"
-	"github.com/supabase/postgrest-go"
 )
 
 const (
 	REST_URL     = "/rest/v1"
 	STORGAGE_URL = "/storage/v1"
+	AUTH_URL     = "/auth/v1"
 )
 
 type Client struct {
+	// Why is this a private field??
 	rest    *postgrest.Client
 	Storage *storage_go.Client
+	// Auth is an interface. We don't need a pointer to an interface.
+	Auth    gotrue.Client
+	options clientOptions
 }
 
-type RestOptions struct {
-	Schema string
-}
-
-type ClientOptions struct {
-	Headers map[string]string
-	Db      *RestOptions
+type clientOptions struct {
+	url     string
+	headers map[string]string
 }
 
 // NewClient creates a new Supabase client.
 // url is the Supabase URL.
 // key is the Supabase API key.
 // options is the Supabase client options.
-func NewClient(url, key string, options *ClientOptions) (*Client, error) {
+func NewClient(url, key string, schema string, headers map[string]string) (*Client, error) {
 	if url == "" || key == "" {
 		return nil, errors.New("url and key are required")
 	}
 
-	headers := map[string]string{
-		"Authorization": "Bearer " + key,
-		"apikey":        key,
+	if headers == nil {
```

Comment:
It seems that the new code is the same as the old one if we continue to use a struct as a parameter.